src/inference/include/openvino/runtime/compiled_model.hpp

```diff
@@ -159,34 +161,52 @@ class OPENVINO_RUNTIME_API CompiledModel {
     void export_model(std::ostream& model_stream);
 
     /**
-     * @brief Sets configuration for current compiled model
-     * @param config Map of pairs: (config parameter name, config parameter value)
+     * @brief Sets properties for current compiled model
+     *
+     * @param property Map of pairs: (property name, property value)
      */
-    void set_config(const ParamMap& config);
+    void set_property(const AnyMap& property);
 
-    /** @brief Gets configuration for a compiled model.
+    /**
+     * @brief Sets properties for current executable network
+     *
+     * @tparam Properties Should be the pack of `std::pair<std::string, ov::Any>` types
+     * @param properties Optional pack of pairs: (property name, property value)
+     * @return nothing
+     */
+    template <typename... Properties>
+    util::EnableIfAllProperties<void, Properties...> set_property(Properties&&... properties) {
+        set_property(AnyMap{std::forward<Properties>(properties)...});
+    }
```

Comment:
Compile-time `RW|RO` check will not be performed here. 
RO property key variable has no method to form `std::pair<std::string, ov::Any>` So it could not be used in set_property exression